rasterio/__init__.py

```diff
@@ -156,11 +159,20 @@ def open(path, mode='r', driver=None, width=None, height=None,
         raise TypeError("invalid driver: {0!r}".format(driver))
     if dtype and not check_dtype(dtype):
         raise TypeError("invalid dtype: {0!r}".format(dtype))
-    if transform:
-        transform = guard_transform(transform)
+
+    if 'affine' in kwargs and transform is not None:
+        raise ValueError("The 'affine' and 'transform' arguments are exclusive")
     elif 'affine' in kwargs:
```

Comment:
@perrygeo if I follow your comments, you are suggesting that lines 163-164 and line 166 get dropped or modified (`if 'transform' not in kwargs: transform = kwargs.pop('affine')`), thus when `affine` is passed in there will be a `DeprecationWarning` (even if passed in w/ transform) and if `transform` is something other than an instance of `Affine`, there will be an exception, right?